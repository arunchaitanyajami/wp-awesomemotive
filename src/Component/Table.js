import { __ } from '@wordpress/i18n'
import * as defaultOptions from '../deafultOptions'

export default ({ tableData, settings }) => {
  const { table } = tableData
  const { data, title } = table
  const { headers, rows } = data
  const { data: { human_date, num_rows, emails } } = settings

  const convertToDate = (timeStamp) => {
    if (!human_date) {
      return timeStamp
    }

    return new Date(timeStamp * 1000).toGMTString()
  }

  return <div className={'awesomemotive-settings-container-fluid'}>
    { table && Object.keys(rows).length > 0 &&
      (
        <div className={'awesomemotive-container-table'}>
          <table cellSpacing={0} cellPadding={0}>
          {title && ( <caption>{__( title, defaultOptions.text_domain )}</caption> )}
          <thead>
            {Object.keys(headers).length > 0 && (
              <tr>
                {Object.keys(headers).map((value, c) => {
                  return <td key={value}>{ __( headers[value], defaultOptions.text_domain ) }</td>
                })}
              </tr>
            )}
          </thead>

          { Object.keys(rows).length > 0 && (
            <tbody>
              {Object.keys(rows.slice(0, num_rows)).map((value, c) => {
                let date = convertToDate(rows[value].date)
                return <tr key={value}>
                  <td>{rows[value].id}</td>
                  <td>{rows[value].url}</td>
                  <td>{rows[value].title}</td>
                  <td>{rows[value].pageviews}</td>
                  <td>{date}</td>
                </tr>
              })}
            </tbody>
          )}
        </table>
        </div>
      )
    }
    <div className={'awesomemotive-container-list'}>
      <h2>{__( 'Emails', defaultOptions.text_domain )}</h2>
    </div>
    <div className={ 'awesomemotive-container-list' }>
      { Object.keys(emails).length > 0 && (
        <ul>
          {Object.keys(emails).map((value, c) => {
            return <li key={value} >{ emails[value] }</li>
          })}
        </ul>
      ) }
    </div>
  </div>
};
