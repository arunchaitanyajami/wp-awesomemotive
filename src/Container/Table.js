import { compose } from '@wordpress/compose';
import { withSelect } from '@wordpress/data';
import Table from '../Component/Table';
import { STORE_NAME } from '../Store/Settings/constants';

const DATA_STORE_NAME = 'awesomemotive/data';
export const mapSelectToProps = ( select ) => {
	const { getData } = select( DATA_STORE_NAME );
	const { getSettings } = select( STORE_NAME );
	return {
		tableData: getData(),
		settings: getSettings(),
	};
};

export default compose( [ withSelect( mapSelectToProps ) ] )( Table );
