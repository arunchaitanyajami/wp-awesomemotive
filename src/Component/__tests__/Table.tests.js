import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Table from '../Table';

const props = {
	tableData: {
		table: {
			title: '',
			data: {
				headers: {},
				rows: {},
			},
		},
	},
	settings: {
		data: {
			human_date: true,
			num_rows: 5,
			emails: {},
		},
	},
};

describe( 'Table Component', () => {
	it( 'Should Render the default component with empty props', async () => {
		const { container } = render( <Table { ...props } /> );
		expect( screen.queryByText( 'table' ) ).not.toBeInTheDocument();
		expect( container ).toMatchSnapshot();
	} );
	it( 'Should Render the component with ordered list of emails', async () => {
		props.settings.data.emails = [ 'test@test.com', 'emial@email.com' ];
		const { container } = render( <Table { ...props } /> );
		expect( container.querySelector( 'ul' ) ).toBeVisible();
		expect( container.querySelector( 'ul > li' ).innerHTML ).toEqual(
			props.settings.data.emails[ 0 ]
		);
		expect( container.querySelectorAll( 'ul li' ).length ).toEqual(
			props.settings.data.emails.length
		);

		expect( container ).toMatchSnapshot();
	} );

	it( 'Should Render the component with table containing data', async () => {
		props.tableData.table.title = 'Testing Table';
		props.tableData.table.data = {
			headers: [ 'ID', 'URL', 'Title', 'Page Views', 'Date' ],
			rows: [
				{
					id: 1,
					url: 'https://google.com',
					title: 'Google co in',
					pageviews: 122,
					date: 1657826894,
				},
				{
					id: 2,
					url: 'https://google2.com',
					title: 'Google 2 co in',
					pageviews: 130,
					date: 1657826894,
				},
			],
		};

		const { container } = render( <Table { ...props } /> );

		expect( container.querySelector( 'table' ) ).toBeVisible();
		expect( container.querySelector( 'table > thead' ) ).toBeVisible();
		expect( container.querySelector( 'table > thead > tr' ) ).toBeVisible();
		expect( container.querySelector( 'table > caption' ) ).toBeVisible();
		expect( container.querySelector( 'table > tbody' ) ).toBeVisible();

		expect(
			container.querySelector( 'table > caption' ).innerHTML
		).toEqual( props.tableData.table.title );
		expect(
			container.querySelector( 'table > thead > tr > td' ).innerHTML
		).toEqual( 'ID' );
		expect(
			container.querySelectorAll( 'table >  thead > tr td' ).length
		).toEqual( props.tableData.table.data.headers.length );

		expect(
			parseInt(
				container.querySelector( 'table > tbody > tr > td' ).innerHTML
			)
		).toEqual( 1 );
		expect(
			container.querySelectorAll( 'table >  tbody > tr' ).length
		).toEqual( 2 );
	} );
} );
