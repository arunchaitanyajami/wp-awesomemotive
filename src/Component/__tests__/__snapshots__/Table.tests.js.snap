// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Should Render the component with ordered list of emails 1`] = `
<div>
  <div
    class="awesomemotive-settings-container-fluid"
  >
    <div
      class="awesomemotive-container-list"
    >
      <h2>
        Emails
      </h2>
    </div>
    <div
      class="awesomemotive-container-list"
    >
      <ul>
        <li>
          test@test.com
        </li>
        <li>
          emial@email.com
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Table Component Should Render the default component with empty props 1`] = `
<div>
  <div
    class="awesomemotive-settings-container-fluid"
  >
    <div
      class="awesomemotive-container-list"
    >
      <h2>
        Emails
      </h2>
    </div>
    <div
      class="awesomemotive-container-list"
    />
  </div>
</div>
`;
